### PR TITLE
UX improve

### PR DESCRIPTION
**Pull request by o-mehlii**
https://trello.com/c/7CDsYO3w/339-improve-ux-for-filter-in-totals-widget

### CODE REVIEW

**Review by a-nahaitsev — CHANGES_REQUESTED** [review]
Is it ok that you didn't follow the instructions?
image

File: renderer/src/components/Totals/Totals.tsx [hunk 1]
```diff
@@ -29,10 +38,7 @@ const Totals = ({ selectedDate }) => {
   const [totals, setTotals] = useState<Total[]>([]);
   const [period, setPeriod] = useState<PeriodWithDate>({
     periodName: "day",
-    date:
-      selectedDate.getDate().toString().padStart(2, "0") +
-      " " +
-      selectedDate.toLocaleDateString("en-US", { month: "short" }),
+    date: day + " " + shortMonth,
```
Comment (a-nahaitsev) [inline renderer/src/components/Totals/Totals.tsx:41]: in the task it was another format

**Review by a-nahaitsev — APPROVED** [review]

File: renderer/src/components/Totals/Totals.tsx [hunk 2]
```diff
@@ -65,6 +78,10 @@ const Totals = ({ selectedDate }) => {
     };
   }, [selectedDate, period]);
 
+  useEffect(() => {
+    changePeriod(period.periodName, selectedDate);
+  }, [selectedDate]);
```
Comment (okhmelevskyi) [inline renderer/src/components/Totals/Totals.tsx:83]: I guess you should add the period as dependencies for useEffect.

File: renderer/src/components/Totals/Totals.tsx [hunk 3]
```diff
@@ -123,9 +140,44 @@ const Totals = ({ selectedDate }) => {
     }
   };
 
+  const changePeriod = (periodName, selectedDate) => {
+    let dateName = "";
+
+    switch (periodName) {
+      case "day":
+        const today = new Date();
```
Comment (okhmelevskyi) [inline renderer/src/components/Totals/Totals.tsx:148]: May I ask you to create a separate function for the code from the 148 line to the 163 line?

File: renderer/src/components/Totals/Totals.tsx [hunk 4]
```diff
@@ -123,9 +140,44 @@ const Totals = ({ selectedDate }) => {
     }
   };
 
+  const changePeriod = (periodName, selectedDate) => {
+    let dateName = "";
+
+    switch (periodName) {
+      case "day":
```
Comment (okhmelevskyi) [inline renderer/src/components/Totals/Totals.tsx:147]: And create a new constant for day, week, today, etc. If it's already existed then use it.
`const DATE_PERIODS = { DAY: 'day' ... };`

File: renderer/src/components/Totals/__tests__/Totals.test.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  send: jest.fn(),
+  invoke: jest.fn(),
+  ...globalIpcRendererMock,
+};
+
+const fakeDate = new Date("2024-03-20T11:30:00");
+const fakeTomorrowDate = new Date("2024-03-21T11:30:00");
+
+describe("Totals date changes", () => {
+  it("displays the correct date text for 20", () => {
```
Comment (okhmelevskyi) [inline renderer/src/components/Totals/__tests__/Totals.test.tsx:32]: Please use this 
```
afterAll(() => {
    jest.restoreAllMocks();

    global.ipcRenderer = globalIpcRendererMock;
  });
```
An example you can see in this file: `renderer/src/components/GoogleConnection/__tests__/GoogleConnection.test.tsx`

**Review by okhmelevskyi — CHANGES_REQUESTED** [review]

File: renderer/src/components/Totals/Totals.tsx (hunk 2 above)
Comment (o-mehlii) [reply renderer/src/components/Totals/Totals.tsx:83]: This causes an infinite recursion. The period is overwritten, the useEffect is triggered, the period is overwritten...

**Review by okhmelevskyi — APPROVED** [review]

**Pull request merged**